### PR TITLE
Update NPC OnSellitem Array List

### PR DESCRIPTION
**Pull request by Emistry**
### Pull Request Prelude

- [X] I have followed [proper Hercules code styling][code].
- [X] I have read and understood the [contribution guidelines][cont] before making this PR.
- [X] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

**Issues addressed:** 
- missing random item option's param array. (Part of #2379)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by Emistry** [comment]
I dont have any good idea for the documentation part.
```
@inventorylist_opt_id1~5[]     - array of random option id.
@inventorylist_opt_val1~5[]    - array of random option val.
@inventorylist_opt_param1~5[]  - array of random option param.
```
any suggestion or advise for me?

File: src/map/npc.c [hunk 1]
```diff
@@ -2764,32 +2769,31 @@ static int npc_selllist_sub(struct map_session_data *sd, struct itemlist *item_l
...
-			// store each of the cards/special info from the item in the array
 			snprintf(card_slot, sizeof(card_slot), "@sold_card%d", j + 1);
-			script->setarray_pc(sd, card_slot, i, (void*)card, &key_card[j]);
+			script->setarray_pc(sd, card_slot, i, (void *)card, &key_card[j]);
 		}
 
 		for (j = 0; j < MAX_ITEM_OPTIONS; j++) {
 			intptr_t opt_idx = item->option[j].index;
 			intptr_t opt_value = item->option[j].value;
+			intptr_t opt_param = item->option[j].value;
```
Comment (4144) [inline src/map/npc.c:2788]: should be ``item->option[j].param``

Comment (Emistry) [reply src/map/npc.c]: fixed.

File: src/map/npc.c [hunk 2]
```diff
@@ -2715,43 +2715,48 @@ static int npc_selllist_sub(struct map_session_data *sd, struct itemlist *item_l
...
-	script->cleararray_pc(sd, "@sold_identify", (void*)0);
+	script->cleararray_pc(sd, "@sold_nameid", (void *)0);
+	script->cleararray_pc(sd, "@sold_quantity", (void *)0);
+	script->cleararray_pc(sd, "@sold_refine", (void *)0);
+	script->cleararray_pc(sd, "@sold_attribute", (void *)0);
+	script->cleararray_pc(sd, "@sold_identify", (void *)0);
 
-	for( j = 0; j < MAX_SLOTS; j++ )
+	for (j = 0; j < MAX_SLOTS; j++)
 	{// clear each of the card slot entries
```
Comment (Kenpachi2k13) [inline src/map/npc.c:2743]:
```suggestion
	for (j = 0; j < MAX_SLOTS; j++) { // Clear each of the card slot entries.
```

Comment (Emistry) [reply src/map/npc.c]: done

**Pull request merged**